qirlib/src/generation/qir/instructions.rs

```diff
@@ -1,41 +1,56 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use super::result;
-use crate::codegen::CodeGenerator;
-use crate::generation::interop::{If, Instruction};
+use crate::{
+    codegen::CodeGenerator,
+    generation::{
+        interop::{Call, If, Instruction, Value},
+        qir::result,
+    },
+};
 use inkwell::values::{BasicValueEnum, FunctionValue, PointerValue};
 use std::collections::HashMap;
 
 /// # Panics
 ///
 /// Panics if the qubit name doesn't exist
 fn get_qubit<'ctx>(
-    name: &str,
     qubits: &HashMap<String, BasicValueEnum<'ctx>>,
+    name: &str,
 ) -> BasicValueEnum<'ctx> {
     *qubits
         .get(name)
         .unwrap_or_else(|| panic!("Qubit {} not found.", name))
 }
 
+fn get_result<'ctx>(
+    generator: &CodeGenerator<'ctx>,
+    results: &HashMap<String, Option<PointerValue<'ctx>>>,
+    name: &str,
+) -> PointerValue<'ctx> {
+    // Panic if an undeclared result name is referenced, and default to zero if the result has been
```

Comment:
Why does this panic instead of returning a `Result` or `Option`?

Reply:
Since results in the Python API are wrapped in an opaque type, the only way to panic is if you create two modules with a different number of results, and pass a `ResultRef` from one into the other. I suppose we should still give a non-panicky exception in that case, but this behavior is from before this PR.

Reply:
Makes sense! Perhaps a TODO comment or an issue to track reducing the number of panic exceptions might be useful, then?

Reply:
I added a todo linking to #31.